app/upgrades/v21/constants.go

```diff
@@ -22,6 +24,8 @@ var Upgrade = upgrades.Upgrade{
 			// v47 modules
 			crisistypes.ModuleName,
 			consensustypes.ModuleName,
+			authenticatortypes.ManagerStoreKey,
```

Comment:
This is temporarily added here for the sake of the tests passing:

If I switch the Makefile to point to v22 for the upgrade version we run into the following error:

```
failed to load latest version: version of store crisis mismatch root store's version; expected 150 got 0; new stores should be added using StoreUpgrades
```

Thoughts welcome here. I'm assuming there is something slightly off with the store wiring or v22 upgrade code?

Reply:
I've seen that error pop up a few times for some people, but not sure anyone knows why 🤔

Reply:
ohh! I think this happens when the current state doesn't match the binary we're running. I think we just have to make sure that the pre-upgrade image and pre-upgrade binary match.

Reply:
Yeah afaik that's the cause. I've parsed through the e2e codebase and from what I tell we need two matching images for the init and the pre upgrade: 

https://github.com/osmosis-labs/osmosis/blob/main/tests/e2e/containers/config.go#L27

https://github.com/osmosis-labs/osmosis/blob/main/tests/e2e/containers/config.go#L29

Reply:
Awesome work here, generated the v22 upgrade handler using a github workflow (linked below), which should update the docker images automatically, I'm happy to merge this into the working branch then fix after we have tagged `v21` and then remove the added changes to the `v21` handler

https://github.com/osmosis-labs/osmosis/pull/7089

Reply:
Sounds good 🤝